pkg/k8s/watchers/callbacks/raw.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package callbacks
+
+import (
+	"github.com/cilium/cilium/pkg/lock"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+// NewRaw creates a new raw callbacks list.
+func NewRaw() *RawList {
+	return &RawList{
+		cbs: make([]cache.ResourceEventHandlerFuncs, 0),
+	}
+}
+
+// Register registers the raw event handler callbacks into the list.
+func (l *RawList) Register(cb cache.ResourceEventHandler) {
+	l.Lock()
+	l.cbs = append(l.cbs, cache.ResourceEventHandlerFuncs{
+		AddFunc:    cb.OnAdd,
+		UpdateFunc: cb.OnUpdate,
+		DeleteFunc: cb.OnDelete,
+	})
+	l.Unlock()
+}
+
+// Callbacks returns all the raw event handler callbacks from the list.
+func (l *RawList) Callbacks() []cache.ResourceEventHandlerFuncs {
+	c := make([]cache.ResourceEventHandlerFuncs, 0)
+	l.RLock()
+	copy(c, l.cbs)
```

Comment:
Nothing will be copied [if `len(l.cbs) == 0`](https://golang.org/pkg/builtin/#copy).
```suggestion
	l.RLock()
	c := make([]cache.ResourceEventHandlerFuncs, len(l.cbs))
	copy(c, l.cbs)
```

Reply:
See earlier comment